lib/options.js

```diff
@@ -174,12 +175,14 @@ exports.parse = function parse(argv) {
     throw new ExitError(fmt('Invalid cluster option: %s', cluster));
   }
 
-  if (options.channel && url.parse(options.channel).protocol) {
+  var protocol = url.parse(options.channel).protocol;
+  if (options.channel && protocol) {
     options.channel = defaults(options.channel, {
       path: 'supervisor-control'
     }, {
-      protocol: 'ws'
+      protocol: protocol,
```

Comment:
doesn't work, because protocol can be `http`, which is why this override section exists. I think `protocol: protocol = 'wss' ? 'wss' : 'ws',` would do the trick.